src/transport/mod.rs

```diff
@@ -42,14 +42,13 @@ pub trait Transport: Send + 'static {
 	type Config: Clone + Default + Send + Sync + 'static;
 
 	/// The type of the read half of the transport.
-	type ReadHalf: for<'a> ReadHalfType<'a, Body = Self::Body>;
+	type ReadHalf<'a>: TransportReadHalf<Body = Self::Body> + 'a;
 
 	/// The type of the write half of the transport.
-	type WriteHalf: for<'a> WriteHalfType<'a, Body = Self::Body>;
+	type WriteHalf<'a>: TransportWriteHalf<Body = Self::Body> + 'a;
 
 	/// Split the transport into a read half and a write half.
-	#[allow(clippy::needless_lifetimes)]
-	fn split<'a>(&'a mut self) -> (<Self::ReadHalf as ReadHalfType<'a>>::ReadHalf, <Self::WriteHalf as WriteHalfType<'a>>::WriteHalf);
+	fn split(&mut self) -> (Self::ReadHalf<'_>, Self::WriteHalf<'_>);
```

Comment:
The RFC mentions the lifetimes are elided so could you just have Self::ReadHalf instead of Self::ReadHalf<'_>? I could definitely be wrong but trying to wrap my head around the concept.

Reply:
RFC then is misguiding or I dont understand it fully  :P